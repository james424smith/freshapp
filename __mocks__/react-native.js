import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
