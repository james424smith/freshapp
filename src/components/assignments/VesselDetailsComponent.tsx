import React from "react";
import { View } from "react-native";
import styles from "../../styles/AssignmentScreenStyles";
import Text from "../StyledText";

type Props = { textLabel: string; textValue: string | number };

const VesselDetailsComponent = (props: Props) => {
  const { textLabel, textValue } = props;
  return (
    <View style={styles.vesselDetails}>
      <Text style={styles.vesselDetailsLabel}>{textLabel}</Text>
      <Text style={styles.vesselDetailsValue}>{textValue}</Text>
    </View>
  );
};

export default VesselDetailsComponent;
