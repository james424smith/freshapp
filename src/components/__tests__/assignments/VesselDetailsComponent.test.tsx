import React from "react";
import { render, cleanup } from "@testing-library/react-native";
import VesselDetailsComponent from "../../assignments/VesselDetailsComponent";

describe("CargoInstallationComponent snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render CargoInstallationComponent correctly as per the ui made ", () => {
    const { toJSON } = render(
      <VesselDetailsComponent
        textLabel={"Some Label"}
        textValue={"Some Value"}
      />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
