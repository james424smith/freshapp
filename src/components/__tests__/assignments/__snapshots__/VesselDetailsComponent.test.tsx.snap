// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CargoInstallationComponent snapshot test should render CargoInstallationComponent correctly as per the ui made  1`] = `
<View
  style={
    Object {
      "flexBasis": "40%",
      "margin": 30,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#2A2E43",
          "fontFamily": "Roboto-Regular",
        },
        Object {
          "color": "#78849E",
          "fontSize": 24,
        },
      ]
    }
  >
    Some Label
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#2A2E43",
          "fontFamily": "Roboto-Regular",
        },
        Object {
          "color": "#2A2E43",
          "fontSize": 26.5,
          "fontWeight": "bold",
        },
      ]
    }
  >
    Some Value
  </Text>
</View>
`;
